### PR TITLE
GPU-based simulation using CUDA, Part 3

### PR DESCRIPTION
**Pull request by utilForever**
This revision includes:
- Support for GPU-based simulation using CUDA (#12)
  - Create classes
    - CUDAParticleSystemSolverBase2
    - CUDAParticleSystemSolverBase3
    - CUDAParticleSystemSolver2
    - CUDAParticleSystemSolver3
    - CUDASPHSolverBase2
    - CUDASPHSolverBase3
    - CUDAPCISPHSolver2
    - CUDAPCISPHSolver3
    - CUDAWCSPHSolver2
    - CUDAWCSPHSolver3
  - Create project 'CUDASPHSim'
    - Create class 'SPHSimExample'
    - Create class 'CUDAPCISPHSolver3Example'
  - Refactor code
    - Remove whitespaces
    - Delete unused variable 'dir'
    - Correct minor typo to 'Vector3D'

### CODE REVIEW

File: Sources/Core/CUDA/CUDAPCISPHSolver2.cpp [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    const SPHSpikyKernel2 kernel(kernelRadius);
+
+    double denom = 0;
+    Vector2D denom1;
+    double denom2 = 0;
+
+    for (size_t i = 0; i < points.Length(); ++i)
+    {
+        const Vector2D& point = points[i];
```
Comment (Snowapril) [inline Sources/Core/CUDA/CUDAPCISPHSolver2.cpp]:
```suggestion
    for (const auto& point : points)
    {
```
I found that index `i` used only for indexing `Array1<Vector2D> points`.
How about replace it with a range-based for loop 😉
Also same things in [`CUDAPCISPHSolver3.cpp`](https://github.com/CubbyFlow/CubbyFlow/pull/117/files#diff-065eebb8c7d8d03b9719a77fba44df304dcaa9b767dee4d96b0e791811630dc7) can be done.

**Review by DveloperY0115 — APPROVED** [review]
Cool! I'm glad that CubbyFlow example now started to support GPU acceleration.
 While handling collision with boundary is still work-in-progress, I hope this will be implemented in the near future. Thank you for your contribution 👍

File: Sources/Core/CUDA/CUDAPCISPHSolver2.cpp (hunk 1 above)
Comment (utilForever) [reply Sources/Core/CUDA/CUDAPCISPHSolver2.cpp]: Resolves in 81dca5e. 🕊️

Comment (Snowapril) [reply Sources/Core/CUDA/CUDAPCISPHSolver2.cpp]: Cool!

**Review by Snowapril — APPROVED** [review]
Totally positive to me! First example of cuda solver in this PR will be great start of other related works(also for `RenderFlow`). 
Thanks for your hard efforts 😊

**Pull request merged**